src/wp-includes/class-wp-block-bindings-registry.php

```diff
@@ -189,20 +189,6 @@ public function register( string $source_name, array $source_properties ) {
 
 		$this->sources[ $source_name ] = $source;
 
-		// Adds `uses_context` defined by block bindings sources.
```

Comment:
There is a single test failure after removing this code:

<img width="826" alt="Screenshot 2024-04-30 at 15 09 16" src="https://github.com/WordPress/wordpress-develop/assets/699132/006d8ec6-6c19-4dcb-b254-1b7cefd9f590">

It's rather expected, so we only need to adjust the test as the block type doesn't change anymore.

Reply:
You're right. I am not sure if that test makes sense anymore. Maybe we can substitute it for a block that consumes merged context from different sources. Something like this but extended: [link](https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/block-bindings/render.php#L124-L158).

If I am not mistaken, this specific test was added to ensure the merge of multiple context done [here](https://github.com/WordPress/wordpress-develop/pull/6456/files#diff-3fd1b8d9e55d4fe09b5f48002d360d3ba9d338d6f33ef3959828424a590b4947L200) was correct. But I think we don't have that problem anymore.

Reply:
I've pushed this commit to illustrate what I mean: [link](https://github.com/WordPress/wordpress-develop/pull/6456/commits/8035804840e6dabe8a102707317491167c65f466).

Reply:
I left a separate comment. In short, I think it's the right direction.